### DISCUSSION TITLE
JSONResponse headers type

### DISCUSSION THREAD

**Discussion by connesy** [encode/starlette#2186] [Potential Issue] [answered]
The `JSONResponse` class parameter `headers` is annotated as `typing.Optional[typing.Dict[str, str]]`, whereas both the base `Response` class and the other sister-classes (`RedirectResponse`, `StreamingResponse` etc) all have `headers` annotated as `typing.Optional[typing.Mapping[str, str]]`.
Is there any reason for `JSONResponse` being the odd one out here?

**Comment by Kludex** [comment]
Probably worth checking when the Mapping were introduced, and see if JSONResponse was forgotten. 🤔

**Reply by euri10** [reply]
5b56e7dc429de946be8c782e0ae69bcc5d5b5585 was latest change, and at the time the JSONResponse headers was already different

**Reply by Kludex** [reply]
I mean the one that introduced `Mapping`

**Reply by euri10** [reply]
2b54f427613ebd1a6ccb2031aa97364e2d5070a5 introduced mapping

**Reply by connesy** [reply, marked as answer]
It seems like the `Mapping` was just forgotten in https://github.com/encode/starlette/commit/2b54f427613ebd1a6ccb2031aa97364e2d5070a5, and then the `dict` annotation was simply updated alongside the others?

**Comment by euri10** [comment]
> It seems like the `Mapping` was just forgotten in [2b54f42](https://github.com/encode/starlette/commit/2b54f427613ebd1a6ccb2031aa97364e2d5070a5), and then the `dict` annotation was simply updated alongside the others?

:+1:

**Reply by connesy** [reply]
Should I make an issue and PR?

**Reply by Kludex** [reply]
You can create a PR, no need for issue. Just link this discussion.

**Reply by connesy** [reply]
PR: https://github.com/encode/starlette/pull/2187
